homeassistant/components/mqtt/client.py

```diff
@@ -190,15 +190,26 @@ async def async_subscribe(
 ) -> CALLBACK_TYPE:
     """Subscribe to an MQTT topic.
 
+    This function is internal to the MQTT integration
+    and may change at any time. It should not be considered
+    a stable API.
+
+    Call the return value to unsubscribe.
+    """
+    return async_subscribe_internal(hass, topic, msg_callback, qos, encoding)
+
+
+def async_subscribe_internal(
```

Comment:
Should we add a callback decorator?

Reply:
yes, good call

Reply:
added in dc55283fdf622cb7d0ad87eb9abd6d6aab8ac744